DNN Platform/Library/Services/FileSystem/Providers/StandardFolderProvider.cs

```diff
@@ -193,13 +192,16 @@ public override string GetFileUrl(IFileInfo file)
         {
             Requires.NotNull("file", file);
 
-            var portalSettings = GetPortalSettings(file.PortalId);
+            var portalSettings = file.PortalId == PortalSettings.Current?.PortalId ?
```

Comment:
Is this thread safe? If not please assign this to variable first than check:

```
var currentPortalSettings = PortalSettings.Current;
var portalSettings = file.PortalId == currentPortalSettings?.PortalId ?
    currentPortalSettings : 
    GetPortalSettings(file.PortalId);
```

Reply:
Yes. It is thread safe. I do not see any problems using it without extra variable.